src/components/Wallet/WalletSpend.jsx

```diff
@@ -83,100 +86,60 @@ class WalletSpend extends React.Component {
     return false;
   };
 
-  signTransaction = () => {
+  showSignTransaction = () => {
     const { setSpendStep } = this.props;
     setSpendStep(SPEND_STEP_SIGN);
   };
 
-  showPreview = () => {
-    const { finalizeOutputs, setSpendStep } = this.props;
+  handleShowPreview = () => {
+    const { autoSelectCoins, autoSpend, finalizeOutputs } = this.props;
+    if (autoSpend) autoSelectCoins();
+    else finalizeOutputs(true);
```

Comment:
manual selection we set outputs to finalized manually when preview is being shown. Otherwise this happens on coin selection. Basically the thinking is that coins are selected, change output is determined, and we should be able to safely say outputs are finalized.